Lab.kt

```diff
@@ -9,88 +9,22 @@ data class ItemData(
     var info: String? = null
 )
 // -----------------------
-
 fun main() {
     val result = processList(listOf(25, "Hola", null, false))
-    println(result)
-    
-}
-
+    println(result) }
 fun processList(inputList: List<Any?>?): List<ItemData>? {
    if (inputList == null) { 
         return null
-    } else {
+   } else {
          var result = ArrayList<ItemData>()
-       
-        for ((indice, item) in inputList.withIndex()) {
-            when (item) {
-              is  String -> result.add(ItemData(
-   	originalPos = indice,
-   	originalValue = item,
-  	type = "cadena",
-  	info = ParaString(item)
-   )
-              )
-              is Boolean -> result.add(ItemData(
-              	originalPos = indice,
-   				originalValue = item,
-  				type = "Booleano",
-  				info = ParaBooleano(item)
-              
-              ))
-              
-              is Int -> result.add(ItemData(
-              
-                originalPos = indice,
-   				originalValue = item,
-  				type = "Entero",
-  				info = ParaEnteros(item)
-              ))
-              
-           else-> 
-               if(item!=null){
-                 result.add(ItemData(
-                originalPos = indice,
-   				originalValue = item,
-  				type = "null",
-  				info = null  
-                   )
-                 )
-               }
-           
-                
-            }
-        
-}    
+        for ((indice, item) in inputList.withIndex()) { 
+ when (item) {
+ is  String -> result.add(ItemData( indice,item,"cadena","L" +item.length.toString()))
```

Comment:
Es bueno tener consistencia en espacios en blanco, hay algunos extras en `is  String` y en `ItemData( indice`.